### PR TITLE
added datatrace config section: fixes #1916

### LINKED ISSUE #1916: Config section compiler_datatrace not defined
When running the compiler, it shows the following warning: `Config section compiler_datatrace not defined`. It seems that the `compiler_datatrace.enable` option is used, but never defined.

### PR DESCRIPTION
**Pull request by sanderr**

### CODE REVIEW

File: src/inmanta/compiler/config.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    "datatrace_enable",
+    False,
+    "Enables the experimental datatrace application on top of the compiler."
+    " The application should help in identifying the cause of compilation errors"
+    " during the development process.",
+    is_bool,
+)
+
+
+def datatrace_enabled() -> bool:
```
Comment (wouterdb) [inline src/inmanta/compiler/config.py]: why wrap it in a method?

Comment (sanderr) [reply src/inmanta/compiler/config.py]: To add a layer of abstraction. If we would want to change the section or the name of this option that shouldn't require changes everywhere this check is performed.

Comment (sanderr) [reply src/inmanta/compiler/config.py]: On second thought, this doesn't make any sense. It's already stored in a variable which provides the desired abstraction. Will fix.

Comment (sanderr) [reply src/inmanta/compiler/config.py]: I've fixed it. I've also moved the check from `inmanta.execute.scheduler` to `inmanta.compiler` which passes it to the `Scheduler` object. There was an import cycle. Also, it makes more sense to check the config in the compiler module.

**Pull request merged**